hpargparse/hputils.py

```diff
@@ -233,15 +236,21 @@ def wrapper(*args, **kwargs):
             parser.add_argument(
                 make_option("list"),
                 action="store",
-                default=False,
+                default=None,
                 const="yaml",
                 nargs="?",
-                choices=["detail", "yaml"],
+                choices=["detail", "yaml", "json"],
                 help=(
                     "List all available hyperparameters. If `{} detail` is"
                     " specified, a verbose table will be print"
                 ).format(make_option("list")),
             )
+        elif action == "detail":
```

Comment:
Because  `--hp-detail` support is added, it’s best for this code to add some explanation in param `inject_actions:` , and test for `--hp-detail` should be added.